src/components/SearchForm.js

```diff
@@ -48,7 +48,7 @@ export default class SearchForm extends React.Component {
     handleSubmit(event) {
         if (this.state.query.length === 0) return;
```

Comment:
I just realized that this function should throw `TypeError: Cannot read property 'state' of null` on the first line since we never do `this.handleSubmit = this.handleSubmit.bind(this);` in the constructor.

Reply:
What do you mean?